File: Classes/GuideStepLine.m
//
//  GuideStepLine.m
//  iFixit
//
//  Created by David Patierno on 8/7/10.
//  Copyright 2010 iFixit. All rights reserved.
//

#import "GuideStepLine.h"


@implementation GuideStepLine

@synthesize lineid, level, bullet, text;

+ (GuideStepLine *)guideStepLineWithDictionary:(NSDictionary *)dict {
	GuideStepLine *guideStepLine = [[GuideStepLine alloc] init];
	guideStepLine.bullet = dict[@"bullet"];
	guideStepLine.level  = [dict[@"level"] integerValue];
	guideStepLine.text   = dict[@"text_rendered"];
	return [guideStepLine autorelease];
}

- (void)dealloc {
    [bullet release];
    [text release];
    
    [super dealloc];
}
@end

File: Classes/GuideStepLine.swift
//
//  GuideStepLine.h
//  iFixit
//
//  Created by David Patierno on 8/7/10.
//  Copyright 2010 iFixit. All rights reserved.
//

import Foundation

@objc
class GuideStepLine: NSObject {
    
    var lineid: Int = 0
    var level: Int = 0
    var bullet: String!
    var text: String!
    
    init(json:[String:AnyObject]) {
        bullet = json["bullet"] as! String
        level  = json["level"] as! Int
        text   = json["text_rendered"] as! String
    }
    
}
